react/features/chat/middleware.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+
+import { CONFERENCE_JOINED } from '../base/conference';
+import { JitsiConferenceEvents } from '../base/lib-jitsi-meet';
+import { playAudio } from '../base/media';
+import { MiddlewareRegistry } from '../base/redux';
+
+import { INCOMING_MSG_SOUND_ID } from './constants';
+
+declare var APP: Object;
+
+/**
+ * Implements the middleware of the chat feature.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case CONFERENCE_JOINED:
+        // XXX Currently there's no need to remove the listener, because
+        // conference instance can not be re-used. Listener will be gone with
+        // the conference instance.
```

Comment:
it is mentioned inside _addChatMsgListener, should I move the check and comment outside of the method or add extra comment ? I don't get the part about listener vs re-using. We're not re-using conference instance on mobile either.

Reply:
I mean that on mobile this may translate into a leak because the don't "reload" the page. We don't reuse the Conference instances though.

Reply:
So it will not translate into a leak. Leak would happen only if we'd try to join again with the same instance. Personally I don't like that approach, but I'm doing it only because we do that everywhere (see /features/base/conference for example).